src/topology/support.lean

```diff
@@ -0,0 +1,178 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Patrick Massot
+-/
+
+import topology.separation
+
+/-!
+# The topological support of a function
+
+In this file we define the topological support of a function `f`, `tsupport f`,
+as the closure of the support of `f`.
+
+Furthermore, we say that `f` has compact support if the topological support of `f` is compact.
+
+## Main definitions
+
+* `function.mul_support`
+* `function.has_compact_mul_support`
+
+## Implementation Notes
+
+* We write all lemmas for multiplicative functions, and use `@[to_additive]` to get the more common
+  additive versions.
+* We do not put the definitions in the `function` namespace, following many other topological
+  definitions that are in the root namespace (compare `embedding` vs `function.embedding`).
+-/
+
+open function set filter
+open_locale topological_space
+
+variables {X α α' β γ δ M E : Type*}
+
+section one
+variables [has_one α]
+variables [topological_space X]
+
+-- todo: add mul_tsupport -> tsupport to to_additive dictionary
+
+/-- The topological support of a function is the closure of its support, i.e. the closure of the
+  set of all elements where the function is not equal to 1. -/
+@[to_additive
+/-" The topological support of a function is the closure of its support. i.e. the closure of the
+  set of all elements where the function is nonzero. "-/]
+def mul_tsupport (f : X → α) : set X := closure (mul_support f)
+
+@[to_additive]
+lemma subset_mul_tsupport (f : X → α) : mul_support f ⊆ mul_tsupport f :=
+subset_closure
+
+@[to_additive]
+lemma is_closed_mul_tsupport (f : X → α) : is_closed (mul_tsupport f) :=
+is_closed_closure
+
+@[to_additive]
+lemma mul_tsupport_eq_empty_iff {f : X → α} : mul_tsupport f = ∅ ↔ f = 1 :=
+by rw [mul_tsupport, closure_empty_iff, mul_support_eq_empty_iff]
+
+@[to_additive]
+lemma image_eq_zero_of_nmem_mul_tsupport {f : X → α} {x : X} (hx : x ∉ mul_tsupport f) : f x = 1 :=
+mul_support_subset_iff'.mp (subset_mul_tsupport f) x hx
+
+end one
+
+section
+
+variables [topological_space α] [topological_space α']
+variables [has_one β] [has_one γ] [has_one δ]
+variables {g : β → γ} {f : α → β} {f₂ : α → γ} {m : β → γ → δ} {x : α}
+
+@[to_additive]
+lemma not_mem_closure_mul_support_iff_eventually_eq : x ∉ mul_tsupport f ↔ f =ᶠ[𝓝 x] 1 :=
+by simp_rw [mul_tsupport, mem_closure_iff_nhds, not_forall, not_nonempty_iff_eq_empty,
+    ← disjoint_iff_inter_eq_empty, disjoint_mul_support_iff, eventually_eq_iff_exists_mem]
+
+/-- A function `f` *has compact multiplicative support* or is *compactly supported* if the closure
+of the multiplicative support of `f` is compact. In a T₂ space this is equivalent to `f` being equal
+to `1` outside a compact set. -/
+@[to_additive
+/-" A function `f` *has compact support* or is *compactly supported* if the closure of the support
+of `f` is compact. In other words: `f` is equal to `0` outside a compact set. "-/]
```

Comment:
Please fix the docstring in the additive case. the multiplicative version is more accurate.